asserts/database.go

```diff
@@ -293,7 +293,7 @@ func (db *Database) Add(assert Assertion) error {
 
 	keyValues := make([]string, len(assertType.PrimaryKey))
 	for i, k := range assertType.PrimaryKey {
-		keyVal := assert.Header(k)
+		keyVal := assert.HeaderString(k)
 		if keyVal == "" {
 			return fmt.Errorf("missing primary key header: %v", k)
```

Comment:
this error message could probably be improved a tiny bit to say "missing or non-string primary key header: %v". might help with debugging.